mrqdlock.hpp

```diff
@@ -1,174 +1,179 @@
-#ifndef qd_mrqdlock_hpp
-#define qd_mrqdlock_hpp qd_mrqdlock_hpp
+#ifndef MRQDLOCK_HPP_
+#define MRQDLOCK_HPP_
 
 #include<array>
 #include<atomic>
+#include<utility>
 #include "readindicator/reader_groups.hpp"
 #include "util/pause.hpp"
 #include "qdlock_base.hpp"
 
-template<class MLock, class DQueue, class RIndicator, int READ_PATIENCE_LIMIT, starvation_policy_t starvation_policy=starvation_policy_t::starvation_free>
-class mrqdlock_impl {
-	char pad1[128];
-	std::atomic<int> writeBarrier;
-	char pad2[128];
-	RIndicator reader_indicator;
-	char pad3[128];
-	typedef qdlock_base<MLock, DQueue, starvation_policy> base;
-	base myqdlock;
-	typedef mrqdlock_impl<MLock, DQueue, RIndicator, READ_PATIENCE_LIMIT, starvation_policy>* this_t;
-	struct reader_indicator_sync {
-		static void wait_writers(base* t) {
-			while(reinterpret_cast<this_t>(t->__data)->writeBarrier.load() > 0) {
-				qd::pause();
+namespace qd {
```

Comment:
My suggestion is to _not_ indent after a namespace.  It causes very long lines and unnecessary differences.